pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -375,6 +281,10 @@ def val_func(v: Any) -> Any:
         # ignore any other unknown metadata
         return None
 
+    if chain_schema_steps:
+        chain_schema_steps = [schema] + chain_schema_steps
+        return cs.chain_schema(chain_schema_steps)
+
```

Comment:
One interesting question here is what happens when you have mixes of constraints that are applied as validator functions vs chain schema additions. This implementation isn't breaking any tests (in fact, fixing bugs), but I could see an argument for considering a pattern for this more carefully.